### ISSUE TITLE
Starts VM only if needed

### ISSUE THREAD

**Issue by holyhope** [Telmate/terraform-provider-proxmox#531]
I use the `proxmox_vm_qemu` to provision a VM, and then transform it to a template.

```terraform
resource "proxmox_vm_qemu" "template" {
  vmid = 100
  // ...

  connection {
    type        = "ssh"
    // ...
  }

  provisioner "remote-exec" {
    inline = [
      "qm stop '${self.vmid}'",
      "qm template '${self.vmid}' -disk scsi0",
    ]
  }
}
```

On next reconciliation, the provider tries to start the vm after changing it, which failed since the vm is now a template.
I suggest to start the VM only if it is the expected state. Example when `hastate` is empty or one of `running` and `enabled`.
WDYT?

**Comment by mleone87** [comment]
Sorry, I don't think I understand, also related to #532  what is the connection between a template and the HA to understand if a vm must remain off or on?

**Comment by kencx** [comment]
Possibly fixed by #725. The [Packer Proxmox builder](https://developer.hashicorp.com/packer/plugins/builders/proxmox) is also a much better way to create templates.
